auto_detect_rpi.py

```diff
@@ -0,0 +1,134 @@
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+'''
+
+import subprocess
+
+# RPI_VARIANTS was inspired from http://www.raspberrypi-spy.co.uk/2012/09/checking-your-raspberry-pi-board-version/
+# This module is meant for retrieving the Raspberry Pi's generation model, PCB model (dimension-wise) and PCB revision
+# Works with Python 3 & 2 !!!
+
+# Each key represents the hardware revision number
+# This isn't the same as the RaspberryPi revision
+# Having the hardware revision number is useful when working with hardware or software.
+
+RPI_VARIANTS = {
+"0002" : ["Model B Rev 1", "RPI1"],
+
+"0003" : ["Model B Rev 1 ECN0001 (no fuses, D14 removed)", "RPI1"],
+
+"0004" : ["Model B Rev 2", "RPI1"],
+"0005" : ["Model B Rev 2", "RPI1"],
+"0006" : ["Model B Rev 2", "RPI1"],
+
+"0007" : ["Model A", "RPI1"],
+"0008" : ["Model A", "RPI1"],
+"0009" : ["Model A", "RPI1"],
+
+"000d" : ["Model B Rev 2", "RPI1"],
+"000e" : ["Model B Rev 2", "RPI1"],
+"000f" : ["Model B Rev 2", "RPI1"],
+
+"0010" : ["Model B+", "RPI1"],
+"0013" : ["Model B+", "RPI1"],
+"900032" : ["Model B+", "RPI1"],
+
+"0011" : ["Compute Module", "RPI-COMPUTE-MODULE"],
+"0014" : ["Compute Module", "RPI-COMPUTE-MODULE"],
+
+"0012" : ["Model A+", "RPI1"],
+"0015" : ["Model A+", "RPI1"],
+
+"a01041" : ["Pi 2 Model B v1.1", "RPI2"],
+"a21041" : ["Pi 2 Model B v1.1", "RPI2"],
+
+"a22042" : ["Pi 2 Model B v1.2", "RPI2"],
+
+"900092" : ["Pi Zero v1.2", "RPI0"],
+
+"900093" : ["Pi Zero v1.3", "RPI0"],
+
+"0x9000C1" : ["Pi Zero W", "RPI0"],
+
+"a02082" : ["Pi 3 Model B", "RPI3"],
+"a22082" : ["Pi 3 Model B", "RPI3"],
+}
+
+# represents indexes for each corresponding key in the above dictionary
+RPI_MODEL_AND_PCBREV = 0
+RPI_GENERATION_MODEL = 1
+
+def getRPIHardwareRevCode():
+    """
+    Returns slightly more descriptive information on the hardware revision of the Raspberry Pi
+    Examples of strings returned : "Model B Rev 2", "Model A+", "Pi 3 Model B", etc
+    Look into the dictionary to see all the possible variants
+    """
+
+    bash_command = "sudo cat /proc/cpuinfo | grep Revision | awk '{print $3}'"
+    revision = sendBashCommand(bash_command)
+    rpi_description = ""
+
+    # if stdout has something
+    if not revision is None:
+        rpi_description = RPI_VARIANTS[revision][RPI_MODEL_AND_PCBREV]
+
+    return rpi_description
+
+def getRPIGenerationCode():
+    """
+    Returns the Raspberry Pi's generation model
+    Depending on the Raspberry Pi's model, the function can return the following strings:
+    "RPI0"
+    "RPI1"
+    "RPi2"
+    "RPI3"
+    "RPI-COMPUTE-MODULE"
+    """
+
+    bash_command = "sudo cat /proc/cpuinfo | grep Revision | awk '{print $3}'"
+    revision = sendBashCommand(bash_command)
+    rpi_description = ""
+
+    # if stdout has something
+    if not revision is None:
```

Comment:
Yeah, but this one is more explicit. This one tells the reader "hey the bash command (`Popen.communicate()`) returns a None - don't forget this!", whereas `if revision` can be anything.

What do you think?